src/commands/image/imageSource/buildImageInAzure/RunStep.ts

```diff
@@ -27,10 +29,21 @@ export class RunStep extends ExecuteActivityOutputStepBase<BuildImageInAzureImag
                 dockerFilePath: path.basename(context.dockerfilePath) /* Assume the dockerfile is always in the root of the source */
             };
 
-            const building: string = localize('buildingImage', 'Building image...');
-            progress.report({ message: building });
+            const retries = 3;
+            await retry(
+                async (currentAttempt: number): Promise<void> => {
+                    const message: string = currentAttempt === 1 ?
+                        localize('buildingImage', 'Building image...') :
+                        localize('buildingImageAttempt', 'Building image (Attempt {0}/{1})...', currentAttempt, retries + 1);
+                    progress.report({ message: message });
+                    ext.outputChannel.appendLog(message);
```

Comment:
How do you feel if we only append log once at the end with a summary of the actions taken rather than sending it the same report updates?  

I think the other log messages in the series follow that format:
<img width="908" alt="image" src="https://github.com/microsoft/vscode-azurecontainerapps/assets/40250218/3d3304c6-0816-415f-a4d8-f67d8786b2f4">

That way the activity log progress can assume responsibility for the continuous "live" updates and the log can output a summary of what happened as well as any other important information that the user may need to resolve issues or gain more info.

Reply:
I'm okay with the live updates in the activity log, but I think appending the log might be important too. If the user is only looking at the output channel, it may appear that the operation froze while it's retrying.

Reply:
Totally cool with showing attempts in the output log.  I think my main issue is against writing out `Building image...` immediately followed by the `Finished building...`.  At a minimum, I would prefer omitting `Building image...` on the first pass so this doesn't happen --  could also maybe play with the `attempt` wording to make sure it matches the style of the rest of the log:

![image](https://github.com/microsoft/vscode-azurecontainerapps/assets/40250218/69103ee3-63b1-48d1-9394-b95632534f16)